internal/cli/cmd/plugin/testschemas/test-schemas.go

```diff
@@ -0,0 +1,531 @@
+		inputPath := filepath.Join(currentPath, "input.json")
+		expectedPath := filepath.Join(currentPath, "expected.json")
+
+		if _, err := os.Stat(inputPath); os.IsNotExist(err) {
+			result.Error = "input.json not found"
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		if _, err := os.Stat(expectedPath); os.IsNotExist(err) {
+			result.Error = "expected.json not found"
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		// Read input data (Grafana format)
+		inputData, err := os.ReadFile(inputPath) //nolint: gosec
+		if err != nil {
+			result.Error = fmt.Sprintf("Failed to read input.json: %v", err)
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		// Read expected data (Perses format)
+		expectedData, err := os.ReadFile(expectedPath) //nolint: gosec
+		if err != nil {
+			result.Error = fmt.Sprintf("Failed to read expected.json: %v", err)
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		// Extract plugin kind from expected.json
+		var expectedPlugin common.Plugin
+		if err := json.Unmarshal(expectedData, &expectedPlugin); err != nil {
+			result.Error = fmt.Sprintf("Failed to parse expected.json: %v", err)
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		// Determine the definition ID and type based on the plugin kind
+		var defID, typeOfData string
+		// Set default definition ID and type based on the plugin kind from the loaded schema
+		switch pluginKind {
+		case v1plugin.KindVariable:
+			defID = "#var"
+			typeOfData = "variable"
+		case v1plugin.KindQuery:
+			defID = "#target"
+			typeOfData = "query"
+		case v1plugin.KindPanel:
+			defID = "#panel"
+			typeOfData = "panel"
+		default:
+			return fmt.Errorf("unsupported migration schema kind: %s", pluginKind)
+		}
+
+		logrus.Debugf("Run migration test for plugin %s of type %s, using definition ID: %s and type: %s",
+			expectedPlugin.Kind, pluginKind, defID, typeOfData)
+		resultPlugin, resultIsEmpty, err := migrate.ExecuteCuelangMigrationScript(buildInstance, inputData, defID, typeOfData)
+		logrus.Debugf("Migration results - success: %v, empty: %v, error: %v", resultPlugin != nil, resultIsEmpty, err)
+
+		if err != nil {
+			result.Error = fmt.Sprintf("Migration execution failed: %v", err)
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		if resultIsEmpty || resultPlugin == nil {
+			result.Error = "Migration resulted in ignored or nil result"
+			results = append(results, result)
+			return filepath.SkipDir
+		}
+
+		// Convert back to map for comparison
+		resultValue := map[string]interface{}{
+			"kind": resultPlugin.Kind,
+			"spec": resultPlugin.Spec,
+		}
+		actualBytes, _ := json.Marshal(resultValue)
+
+		// Compare JSON outputs (normalized)
+		var actualNormalized, expectedNormalized interface{}
+		if err := json.Unmarshal(actualBytes, &actualNormalized); err != nil {
```

Comment:
same for the actualNormalized, I don't see the value of Unmarshalling it into a generic struct since you already have the data in memory in a slice of byte